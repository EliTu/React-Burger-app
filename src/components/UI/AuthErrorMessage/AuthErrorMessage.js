import React from 'react';
import styles from './AuthErrorMessage.module.css';
import PropTypes from 'prop-types';

const AuthErrorMessage = ({ errorMessage }) => {
	// CSS Modules Styles
	const { AuthErrorMessage } = styles;

	let message;
	switch (errorMessage) {
		case 'EMAIL_EXISTS':
			message = `The email you've provided is already taken. Please provide a different email address`;
			break;

		case 'TOO_MANY_ATTEMPTS_TRY_LATER':
			message = 'Too many attempts. Please try again later';
			break;

		case 'EMAIL_NOT_FOUND':
			message = `The email you've provided does not exist. Please try signing in with a different email or register a new user with the unused email`;
			break;

		case 'INVALID_PASSWORD':
			message = 'Wrong password. Please try again';
			break;

		case 'USER_DISABLED':
			message = `The account you're trying to sign in to has been disabled`;
			break;

		default:
			message = 'Oops! an Unexpected error has occurred!';
	}

	return (
		<>
			<p className={AuthErrorMessage}>{message}</p>
		</>
	);
};

AuthErrorMessage.propTypes = {
	errorMessage: PropTypes.string,
	authType: PropTypes.string,
};

export default AuthErrorMessage;
