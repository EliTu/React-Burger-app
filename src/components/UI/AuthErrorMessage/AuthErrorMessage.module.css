.AuthErrorMessage {
    width: 90%;
    padding: 0.1rem 0.1rem;
    color: rgb(240, 94, 94);
    margin: 0.2rem auto;
    font-size: 0.9rem;
}