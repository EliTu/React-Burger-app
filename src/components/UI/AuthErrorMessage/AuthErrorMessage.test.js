import React from 'react';
import AuthErrorMessage from './AuthErrorMessage';
import { shallow } from 'enzyme';

describe('<AuthErrorMessage>', () => {
	let component = shallow(<AuthErrorMessage />);

	it('should render the component without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should render the correct message corresponding to EMAIL_EXISTS error', () => {
		component.setProps({ errorMessage: 'EMAIL_EXISTS' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toEqual(
			`The email you've provided is already taken. Please provide a different email address`
		);
	});

	it('should render the correct message corresponding to TOO_MANY_ATTEMPTS_TRY_LATER error', () => {
		component.setProps({ errorMessage: 'TOO_MANY_ATTEMPTS_TRY_LATER' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toEqual(
			`Too many attempts. Please try again later`
		);
	});

	it('should render the correct message corresponding to EMAIL_NOT_FOUND error', () => {
		component.setProps({ errorMessage: 'EMAIL_NOT_FOUND' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toEqual(
			`The email you've provided does not exist. Please try signing in with a different email or register a new user with the unused email`
		);
	});

	it('should render the correct message corresponding to INVALID_PASSWORD error', () => {
		component.setProps({ errorMessage: 'INVALID_PASSWORD' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toEqual(
			`Wrong password. Please try again`
		);
	});

	it('should render the correct message corresponding to USER_DISABLED error', () => {
		component.setProps({ errorMessage: 'USER_DISABLED' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toEqual(
			`The account you're trying to sign in to has been disabled`
		);
	});

	it('should render a default error message if errorMessage has no value', () => {
		component.setProps({ errorMessage: '' });

		expect(component).toMatchSnapshot();
		expect(component.find('p').text()).toEqual(
			`Oops! an Unexpected error has occurred!`
		);
	});
});
