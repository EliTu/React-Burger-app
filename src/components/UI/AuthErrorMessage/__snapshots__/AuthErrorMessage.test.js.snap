// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthErrorMessage> should render a default error message if errorMessage has no value 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    Oops! an Unexpected error has occurred!
  </p>
</Fragment>
`;

exports[`<AuthErrorMessage> should render the component without errors 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    Oops! an Unexpected error has occurred!
  </p>
</Fragment>
`;

exports[`<AuthErrorMessage> should render the correct message corresponding to EMAIL_EXISTS error 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    The email you've provided is already taken. Please provide a different email address
  </p>
</Fragment>
`;

exports[`<AuthErrorMessage> should render the correct message corresponding to EMAIL_NOT_FOUND error 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    The email you've provided does not exist. Please try signing in with a different email or register a new user with the unused email
  </p>
</Fragment>
`;

exports[`<AuthErrorMessage> should render the correct message corresponding to INVALID_PASSWORD error 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    Wrong password. Please try again
  </p>
</Fragment>
`;

exports[`<AuthErrorMessage> should render the correct message corresponding to TOO_MANY_ATTEMPTS_TRY_LATER error 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    Too many attempts. Please try again later
  </p>
</Fragment>
`;

exports[`<AuthErrorMessage> should render the correct message corresponding to USER_DISABLED error 1`] = `
<Fragment>
  <p
    className="AuthErrorMessage"
  >
    The account you're trying to sign in to has been disabled
  </p>
</Fragment>
`;
